diff --git a/version.php b/version.php
index 3cf35e5..ea88c2a 100644
--- a/version.php
+++ b/version.php
@@ -6,10 +6,10 @@
 // This is compared against the values stored in the database to determine
 // whether upgrades should be performed (see lib/db/*.php)
 
-    $version = 2007101592.01; // YYYYMMDD      = date of the 1.9 branch (don't change)
+    $version = 2007101591.16; // YYYYMMDD      = date of the 1.9 branch (don't change)
                               //         X     = release number 1.9.[0,1,2,3,4,5...]
                               //          Y.YY = micro-increments between releases
 
-    $release = '1.9.19+ (Build: 20120831)';     // Human-friendly version name
+    $release = '1.9.18 (Build: 20120514)';     // Human-friendly version name
 
 ?>
