diff --git a/elis/core/lib/filtering/simpleselect.php b/elis/core/lib/filtering/simpleselect.php
index 4b2bd80..08813a9 100644
--- a/elis/core/lib/filtering/simpleselect.php
+++ b/elis/core/lib/filtering/simpleselect.php
@@ -101,6 +101,11 @@ class generalized_filter_simpleselect extends generalized_filter_type {
 
             // Have to do this manually because php function renumber numeric indexes.
             foreach ($this->_options as $key => $value) {
+                // ELIS-7034 -- Menu select options are including a carriage return at the end of the $key property, causing
+                //              the filter to not return any data on form submission. =(
+                if (substr($key, -1) == "\n" || substr($key, -1) == "\r") {
+                    $key = substr($key, 0, -1);
+                }
                 $choices[$key] = $value;
             }
         }
