diff --git a/elis/core/lib/form/gradebookidnumber.php b/elis/core/lib/form/gradebookidnumber.php
index 4fd08a8..72b0c8c 100644
--- a/elis/core/lib/form/gradebookidnumber.php
+++ b/elis/core/lib/form/gradebookidnumber.php
@@ -3,7 +3,7 @@
  * Select an ID Number for a gradebook item from a Moodle course
  *
  * ELIS(TM): Enterprise Learning Intelligence Suite
- * Copyright (C) 2008-2011 Remote-Learner.net Inc (http://www.remote-learner.net)
+ * Copyright (C) 2008-2012 Remote Learner.net Inc http://www.remote-learner.net
  *
  * This program is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -22,7 +22,7 @@
  * @subpackage core
  * @author     Remote-Learner.net Inc
  * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL
- * @copyright  (C) 2008-2011 Remote Learner.net Inc http://www.remote-learner.net
+ * @copyright  (C) 2008-2012 Remote Learner.net Inc http://www.remote-learner.net
  *
  */
 
