diff --git a/user/filters/user_filter_forms.php b/user/filters/user_filter_forms.php
index 9cd5079..a16be2e 100644
--- a/user/filters/user_filter_forms.php
+++ b/user/filters/user_filter_forms.php
@@ -1,4 +1,4 @@
-<?php //$Id$
+<?php
 
 require_once($CFG->libdir.'/formslib.php');
 
@@ -29,6 +29,17 @@ class user_add_filter_form extends moodleform {
         // Don't use last advanced state
         $mform->setShowAdvanced(false);
     }
+
+    /**
+     * Obtain the list of elements that that one or more elements depend on
+     *
+     * @return  array  List of the names of the appropriate elements
+     */
+    function get_dependencies() {
+        //fetch from QuickForm
+        return $this->_form->_dependencies;
+    }
+
 }
 
 class user_active_filter_form extends moodleform {
