diff --git a/elis/core/plugins/user_activity/etl.php b/elis/core/plugins/user_activity/etl.php
index 89b276d..d8e1802 100644
--- a/elis/core/plugins/user_activity/etl.php
+++ b/elis/core/plugins/user_activity/etl.php
@@ -127,13 +127,17 @@ function user_module_activity_add_session($userid, $courseid, $cmid, $session_st
         $first = true;
         while ($session_end > $start_hour + 3600) {
             $session_hour_duration = $start_hour + 3600 - $session_start;
-            if ($first && $rec = $DB->get_record(ETL_MOD_TABLE,
+            if ($rec = $DB->get_record(ETL_MOD_TABLE,
                                          array('userid' => $userid,
                                                'cmid'   => $cmid,
                                                'hour'   => $start_hour))) {
                 $rec->duration += $session_hour_duration;
+                if ($rec->duration <= 3600) {
                 $DB->update_record(ETL_MOD_TABLE, $rec);
             } else {
+                    mtrace("\nuser_module_activity_add_session(userid = {$userid}, courseid = {$courseid}, cmid = {$cmid}, session_start = {$session_start}, session_end = {$session_end}): Warning: duration > 3600\n");
+                }
+            } else {
                 $rec = new stdClass;
                 $rec->userid = $userid;
                 $rec->courseid = $courseid;
@@ -147,13 +151,17 @@ function user_module_activity_add_session($userid, $courseid, $cmid, $session_st
             $first = false;
         }
         $remainder = $session_end - $session_start;
-        if ($first && $rec = $DB->get_record(ETL_MOD_TABLE,
+        if ($rec = $DB->get_record(ETL_MOD_TABLE,
                                       array('userid' => $userid,
                                             'cmid'   => $cmid,
                                             'hour'   => $start_hour))) {
             $rec->duration += $remainder;
+            if ($rec->duration <= 3600) {
             $DB->update_record(ETL_MOD_TABLE, $rec);
         } else {
+                mtrace("\nuser_module_activity_add_session(userid = {$userid}, courseid = {$courseid}, cmid = {$cmid}, session_start = {$session_start}, session_end = {$session_end}): Warning: duration > 3600\n");
+            }
+        } else {
             $rec = new stdClass;
             $rec->userid = $userid;
             $rec->courseid = $courseid;
@@ -245,8 +253,9 @@ function user_activity_task_process(&$state) {
 
     // find the last record that's close to our chunk size, without
     // splitting a second between runs
-    $endtime = $DB->get_field_select('log', 'MIN(time)', 'id >= ?',
-                                     array($startrec + USERACT_RECORD_CHUNK));
+    $endtime = $DB->get_field_select('log', 'MIN(time)', 'id >= ? AND time > ?',
+                                     array($startrec + USERACT_RECORD_CHUNK,
+                                           $starttime));
     if (!$endtime) {
         $endtime = time();
     }
