diff --git a/lib/environmentlib.php b/lib/environmentlib.php
index 067b9cb..d547df7 100644
--- a/lib/environmentlib.php
+++ b/lib/environmentlib.php
@@ -482,17 +482,6 @@ function environment_check_php_extensions($version) {
         $result->setErrorCode(NO_PHP_EXTENSIONS_SECTION_FOUND);
         return array($result);
     }
-
-/// Hack alert: inject extra mysqli dependency, mysql PHP extension is not supported in Moodle 2.x, warn only ppl using MySQL.
-    global $CFG;
-    set_dbfamily();
-    if ($CFG->dbfamily === 'mysql') {
-        $mver = normalize_version($version);
-        if (version_compare($mver, '2.0', 'ge')) {
-            $data['#']['PHP_EXTENSIONS']['0']['#']['PHP_EXTENSION'][] = array('#' => '', '@' => array('name' => 'mysqli', 'level' => 'required'));
-        }
-    }
-
 /// Iterate over extensions checking them and creating the needed environment_results
     foreach($data['#']['PHP_EXTENSIONS']['0']['#']['PHP_EXTENSION'] as $extension) {
         $result = new environment_results('php_extension');
