diff --git a/admin/renderer.php b/admin/renderer.php
index 13f3a08..2d68add 100644
--- a/admin/renderer.php
+++ b/admin/renderer.php
@@ -470,7 +470,7 @@ class core_admin_renderer extends plugin_renderer_base {
         $copyrighttext = '<a href="http://moodle.org/">Moodle</a> '.
                          '<a href="http://docs.moodle.org/dev/Releases" title="'.$CFG->version.'">'.$CFG->release.'</a><br />'.
                          'Copyright &copy; 1999 onwards, Martin Dougiamas<br />'.
-                         'and <a href="http://docs.moodle.org/dev/Credits">many other contributors</a>.<br />'.
+                         'and <a href="http://moodle.org/dev">many other contributors</a>.<br />'.
                          '<a href="http://docs.moodle.org/dev/License">GNU Public License</a>';
         //////////////////////////////////////////////////////////////////////////////////////////////////
 
@@ -971,6 +971,7 @@ class core_admin_renderer extends plugin_renderer_base {
 
                 // ELIS_2.0.0 Release
                 $versionfile = $plugin->rootdir .'/version.php';
+                $module = new stdClass; // ELIS-7928: was getting previous value
                 if (file_exists($versionfile)) {
                     include($versionfile);
                 }
